libparsec/crates/platform_mountpoint/src/windows/mod.rs

```diff
@@ -0,0 +1,481 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+mod error;
+
+use anyhow::anyhow;
+use once_cell::sync::{Lazy, OnceCell};
+use std::{
+    ops::Deref,
+    path::{Path, PathBuf},
+    sync::{Arc, Mutex},
+};
+
+use winfsp_wrs::{
+    filetime_from_utc, filetime_now, u16cstr, CleanupFlags, CreateFileInfo, CreateOptions,
+    FileAccessRights, FileAttributes, FileInfo, FileSystem, FileSystemContext, PSecurityDescriptor,
+    Params, SecurityDescriptor, U16CStr, U16CString, VolumeInfo, VolumeParams,
+    WriteMode as WinWriteMode, NTSTATUS, STATUS_DIRECTORY_NOT_EMPTY, STATUS_NOT_IMPLEMENTED,
+    STATUS_RESOURCEMANAGER_READ_ONLY,
+};
+
+use libparsec_types::FileDescriptor;
+
+use crate::{EntryInfo, EntryInfoType, MountpointInterface, MountpointManager, WriteMode};
+
+static SD: Lazy<SecurityDescriptor> = Lazy::new(|| {
+    SecurityDescriptor::from_wstr(u16cstr!("O:BAG:BAD:P(A;;FA;;;SY)(A;;FA;;;BA)(A;;FA;;;WD)"))
+        .expect("unreachable, valid SecurityDescriptor")
+});
+
+static VOLUME: OnceCell<Arc<Mutex<VolumeInfo>>> = OnceCell::new();
+
+const ENTRY_INFO_EXTENSION: &str = "__parsec_entry_info__";
+const MAX_FILE_NODES: u64 = 1024;
+const MAX_FILE_SIZE: u64 = 16 * 1024 * 1024;
+const ALLOCATION_UNIT: u64 = 4096;
+const FILE_NODES: u64 = 1;
+const VOLUME_LABEL: &U16CStr = u16cstr!("parsec");
+
+fn is_entry_info_path(path: &Path) -> bool {
+    path.extension()
+        .map(|ext| ext == ENTRY_INFO_EXTENSION)
+        .unwrap_or_default()
+}
+
+impl From<&EntryInfo> for FileInfo {
+    fn from(value: &EntryInfo) -> Self {
+        let created = filetime_from_utc(value.created);
+        let updated = filetime_from_utc(value.updated);
+
+        match value.ty {
+            EntryInfoType::File => *FileInfo::default()
+                .set_file_attributes(value.ty.into())
+                .set_creation_time(created)
+                .set_last_access_time(updated)
+                .set_last_write_time(updated)
+                .set_change_time(updated)
+                .set_index_number(value.id.as_u128() as u64)
+                .set_file_size(value.size)
+                .set_allocation_size(
+                    ((value.size + ALLOCATION_UNIT - 1) / ALLOCATION_UNIT) * ALLOCATION_UNIT,
+                ),
+            EntryInfoType::Dir => *FileInfo::default()
+                .set_file_attributes(value.ty.into())
+                .set_creation_time(created)
+                .set_last_access_time(updated)
+                .set_last_write_time(updated)
+                .set_change_time(updated)
+                .set_index_number(value.id.as_u128() as u64),
+            EntryInfoType::ParsecEntryInfo => *FileInfo::default()
+                .set_file_attributes(value.ty.into())
+                // Arbitrary non-zero size
+                .set_file_size(1024)
```

Comment:
Why not a zero size ? having an empty file is something very common :/

Reply:
From `2.16` comment

Reply:
Can you add a `// TODO: should we use 0 value instead ?`, we will then try to tweak this once the FS is working fine ^^